src/session/content/chat_history.rs

```diff
@@ -225,6 +232,37 @@ impl ChatHistory {
         }
     }
 
+    fn leave_chat(&self) {
+        if let Some(chat) = self.chat() {
+            let dialog = adw::MessageDialog::new(
+                Some(&self.parent_window().unwrap()),
+                None,
+                Some("Do you want to leave this chat?"),
+            );
+            dialog.add_responses(&[("no", &gettext("_No")), ("yes", &gettext("_Yes"))]);
+            dialog.set_default_response(Some("no"));
+            dialog.set_close_response("no");
+            dialog.set_response_appearance("yes", adw::ResponseAppearance::Destructive);
+            dialog.run_async(
```

Comment:
I'd make this use `run_future` and I'd make the entire `show_leave_chat_dialog` function async. This way the `spawn` below can be removed.

It should look something like:

```rust
let response = dialog.run_future(None).await;
dialog.close();

let result = ...
```